### PR TITLE
Migrate CircleCI build-and-test workflow to GitHub Actions (1/3)

### PR DESCRIPTION
**Pull request by AzfaarQureshi**
## Which problem is this solving?

As part of https://github.com/open-telemetry/opentelemetry-collector/issues/1234, this pull request migrates the windows-test, setup-environment, lint, and cross-compile workflow that was contained in the config.yml file to GitHub Actions. The remaining CI jobs will be migrated in PR 2/3 and the publish jobs will be migrated in 3/3

## Migration Plan

We suggest having CircleCI and GitHub Action jobs run in parallel for a few weeks. After the GitHub Actions jobs are running fine for a week or so and then remove the CircleCI workflows from config.yml
cc- @alolita, @shovnik

### CODE REVIEW

File: .github/workflows/check-links/check-links.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+set -eu
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+REPO_DIR="$( cd "${SCRIPT_DIR}/../../../" && pwd )"
+GITHUB_REF=${GITHUB_REF:-}
+
+diff_files="$( git diff HEAD origin/master --name-only )"
+check_all_files=1
+if [[ "$GITHUB_REF" = "ref/heads/master" ]] || [[ -n "$( echo "$diff_files" | grep ".github/workflows/check-links" )" ]]; then
```
Comment (AzfaarQureshi) [inline .github/workflows/check-links/check-links.sh:25]: copying over the contents of `.circleci/check-links/`. Needed to tweak this script so it works with the GitHub Actions environment variables and container structure

File: .github/workflows/check-links/config.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+    "ignorePatterns": [
+        {
+            "pattern": "http(s)?://\\d+\\.\\d+\\.\\d+\\.\\d+"
```
Comment (AzfaarQureshi) [inline .github/workflows/check-links/config.json:4]: copy pasted from .circleci/check-links

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      - name: Setup Node
+        uses: actions/setup-node@v2-beta
+        with:
+          node-version: '9'
+      - name: Check links in markdown docs
+        run: |
+          pushd $HOME
+          npm install --save-dev markdown-link-check@3.8.1
+          popd
+      - run: ./.github/workflows/check-links/check-links.sh
```
Comment (bogdandrutu) [inline .github/workflows/build-and-test.yml:55]: I would move this in a separate action. This should not block a release or any testing/releasing steps, only merging :)

Comment (AzfaarQureshi) [reply .github/workflows/build-and-test.yml:55]: oh ok i see. So `check_links` should only be run on an open pull request?

Comment (bogdandrutu) [reply .github/workflows/build-and-test.yml:55]: Can be run on master as well, but should not block other actions if possible

**Comment by AzfaarQureshi** [comment]
@bogdandrutu I've separated out `check-links` into its own workflow so it doesn't block `build-and-test` :)

**Comment by AzfaarQureshi** [comment]
(its failing contrib-tests but @tigrannajaryan  says thats not a blocker for collector-core PRs)

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**